client/components/details-link/index.tsx

```diff
@@ -3,18 +3,24 @@
 /**
  * External dependencies
  */
-import Gridicon from 'gridicons';
+import * as React from 'react';
+import * as Gridicon from 'gridicons';
 import { addQueryArgs } from '@wordpress/url';
 import { Link } from '@woocommerce/components';
 
-export const getDetailsURL = ( id, parentSegment ) =>
+export const getDetailsURL = ( id: string, parentSegment: string ) =>
```

Comment:
With rule `@typescript-eslint/explicit-module-boundary-types` on, linter complains this exported function is missing return type.

```suggestion
export const getDetailsURL = ( id: string, parentSegment: string ): string =>
```